include/kernel_structs.h

```diff
@@ -131,7 +132,17 @@ struct _cpu {
 #endif
 
 #ifdef CONFIG_SCHED_THREAD_USAGE
+	/*
+	 * [usage0] is used as a timestamp to mark the beginning of an
+	 * execution window. [0] is a special value indicating that it
+	 * has been stopped (but not disabled).
+	 * [track_usage] is used to enable/disable this feature.
+	 */
+
+	uint8_t	 track_usage;
```

Comment:
Hm... I guess I'm a little confused what the design here is.  I see what it does: track_usage gates the collection of thread data on a specific CPU.  I just don't see what the point is?  Why would you want to control this per-CPU and not per-thread?

If it's just to avoid performance impact on those other CPUs, then let's fix the impact (some notes elsewhere to that effect).

If it's because you only expect certain threads to run on certain CPUs, then... this seems like a footgun to me.  There is already a k_thread_cpu_mask_*() API to do thread pinning.  Duplicating it in a different way here opens up the possibility of missing data from a thread that should have been pinned but wasn't, or conversely getting a surprising zero out of your tracking because you forgot to enable the right CPU, etc...

Reply:
I was thinking that per thread management may lead to it being too fine-grained and thus a pain to manage from an app developer perspective, and thought that per-core might be a nice balance. As I ponder this more now, I think that I should have though it through more clearly before.

Reply:
I've reworked this one (will upload soon) so that there are two enable/disable pairs--one for gathering thread stats and the other for gathering the CPU stats. Flipping one should not impact the other.